common/metrics/otel_metrics_handler.go

```diff
@@ -87,30 +113,61 @@ func (omp *otelMetricsHandler) Counter(counter string) CounterIface {
 	}
 
 	return CounterFunc(func(i int64, t ...Tag) {
-		option := metric.WithAttributes(tagsToAttributes(omp.tags, t, omp.excludeTags)...)
+		option := metric.WithAttributeSet(omp.makeSet(t))
 		c.Add(context.Background(), i, option)
 	})
 }
 
+func (omp *otelMetricsHandler) getGaugeAdapter(gauge string) (*gaugeAdapter, error) {
+	if v, ok := omp.gauges.Load(gauge); ok {
+		return v.(*gaugeAdapter), nil
+	}
+	adapter := &gaugeAdapter{
+		values: make(map[attribute.Distinct]gaugeValue),
+	}
+	if v, wasLoaded := omp.gauges.LoadOrStore(gauge, adapter); wasLoaded {
+		return v.(*gaugeAdapter), nil
```

Comment:
Is there a race condition here? If thread 1 stores a new gauge here, and then thread 2 hits line 123 and uses the gauge before thread 1 executes line 135, what happens?

Reply:
Good question. It's fine because thread 2 will just store values in the adapter and then they'll get read out by the callback, same as if they happened in a different order. The only issue would be if calling `Float64ObservableGauge` fails, then the adapter would be abandoned.

Digging into the otel library... it looks like it basically can't fail to create a new basic gauge. The only failures seem to come from incorrectly configured aggregation, which is an internal otel thing and would indicate a bug there.